swarm.go

```diff
@@ -1,19 +1,74 @@
 package config
 
 type SwarmConfig struct {
-	AddrFilters             []string
+	// AddrFilters specifies a set libp2p addresses that we should never
+	// dial or receive connections from.
+	AddrFilters []string
+
+	// DisableBandwidthMetrics disables recording of bandwidth metrics for a
+	// slight reduction in memory usage. You probably don't need to set this
+	// flag.
 	DisableBandwidthMetrics bool
-	DisableNatPortMap       bool
-	DisableRelay            bool
-	EnableRelayHop          bool
 
-	// autorelay functionality
-	// if true, then the libp2p host will be constructed with autorelay functionality.
+	// DisableNatPortMap turns off NAT port mapping (UPnP, etc.).
+	DisableNatPortMap bool
+
+	// DisableRelay explicitly disables the relay transport.
+	//
+	// Deprecated: This flag is deprecated and is overridden by
+	// `Transports.Relay` if specified.
+	DisableRelay bool `json:",omitempty"`
+
+	// EnableRelayHop makes this node act as a public relay, relaying
+	// traffic between other nodes.
+	EnableRelayHop bool
+
+	// EnableAutoRelay enables the "auto relay" feature.
+	//
+	// When both EnableAutoRelay and RelayHop are set, this go-ipfs node
+	// will advertise itself as a public relay, instead of finding and using
+	// advertised public relays.
```

Comment:
```suggestion
	// When both EnableAutoRelay and EnableRelayHop are set, this go-ipfs node
	// will advertise itself as a public relay. Otherwise it will find and use
	// advertised public relays.
```

Reply:
is this and "may use" advertised public relays? We won't use the relays if AutoNAT tells us we are dialable, right?